### ISSUE TITLE
Strange results when t_eval has repeat values

### ISSUE THREAD

**Issue by TomWagg** [Nicholaswogan/numblsoda#7]
Thanks for the great package!

**Summary:**
I'm trying to switch over from using scipy's `odeint` to NumbaLSODA as it seems much better! Sometimes the `t_eval` array I supply has repeated times at the end and this produces some strange output. This was okay with odeint though, would it be possible to allow it for NumbaLSODA too? I think it is trying to integrate an infinitely small timestep rather than just evaluating the same timestep multiple times.

**Details:**
I'm trying to add this to [LEGWORK](https://legwork.readthedocs.io/en/latest/) so that we can speed up how we evolve the orbits of binary stars due to gravitational wave emission. However, if the evolution gets too close to the merger then LSODA produces a bunch of warnings about the timesteps being too small (since the derivatives get so large). To avoid this, I set any timesteps that are too close to the merger equal to the last allowable timestep - this leads to repeated values and the weird results that I see. (I don't just trim the array since I do this in a 2D array with a collection of binaries and so each individual binary might have a different length array which would wouldn't work in a 2D array I think.)

**MWE:**
Evolve binary evolution due to gravitational wave emission using Peters (1964)
```
from numba import cfunc
import numba as nb
import numpy as np
from NumbaLSODA import lsoda_sig, lsoda

@cfunc(lsoda_sig)
def de_dt(t, e, de, data):
    e_ = nb.carray(e, (1,))
    beta, c_0 = nb.carray(data, (2,))
    de[0] = -19 / 12 * beta / c_0**4 * (e_[0]**(-29.0 / 19.0) * (1 - e_[0]**2)**(3.0/2.0)) \
        / (1 + (121./304.) * e_[0]**2)**(1181./2299.)

ecc = 0.5
beta = 2.47e22
c_0 = 1.677e10
t_merge = 1.81e17

funcptr = de_dt.address
u0 = np.array([ecc])
data = np.array([beta, c_0])
t_eval = np.array([0.0, 0.1, 0.2, 0.5, 0.8, 0.9, 0.9, 0.9]) * t_merge

ecc_evol, _ = lsoda(funcptr, u0, t_eval=t_eval, data=data)
print(ecc_evol)
```

This outputs
```
[[ 5.00000000e-01]
 [ 4.83244343e-01]
 [ 4.64840518e-01]
 [ 3.95592532e-01]
 [ 2.82404708e-01]
 [ 2.15245963e-01]
 [-1.36311572e+57]
 [-1.36311572e+57]]
```
So you can see that as soon as the timesteps repeat, you get this hugely small number (and eccentricity should be between 0 and 1)

**Comment by TomWagg** [comment]
Oh and also hi @Nicholaswogan, I just started as a grad student at UW astronomy, maybe I'll see you in a meeting at some point 😄

**Comment by Nicholaswogan** [comment]
@TomWagg Thanks for the issue. I think i fixed it: 7748fdf549cb11fa994527485d520d1f0d330eda . Your test case now produces

```
[[0.5       ]
 [0.48324434]
 [0.46484052]
 [0.39559253]
 [0.28240471]
 [0.21524596]
 [0.21524596]
 [0.21524596]]
```

This must be right? Once you confirm I will publish a new version on pypi.

Welcome to UW! Ya, I'm sure we will see each other. I hang out with Vikki Meadows group pretty often. LEGWORK looks awesome.

**Comment by TomWagg** [comment]
@Nicholaswogan awesome, this looks perfect, thank you for fixing so quickly!!

Thanks! Ah nice, she seems great! 😁 Thank you, glad you like it ☺️
